bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/ICloudSession.java

```diff
@@ -137,7 +137,8 @@ private String request(String method, String url, @Nullable String body,
 
         HttpResponse<?> response = this.client.send(request, BodyHandlers.ofString());
 
-        String responseBodyAsString = response.body() != null ? response.body().toString() : "";
+        Object responseBody = response.body();
+        String responseBodyAsString = responseBody != null ? responseBody.toString() : "";
```

Comment:
Why is this change needed? Response is a local value, so no other thread could have modified it in between...

Reply:
Response.body() is nullable.